src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/TestClasses.cs

```diff
@@ -158,4 +158,52 @@ public struct PersonStruct
         public string FirstName { get; set; }
         public string LastName { get; set; }
     }
+
+
+
+    public class CovariantBase
+    {
+        [JsonIgnore]
+        public virtual object Id { get; }
+    }
+    public class CovariantDerived : CovariantBase
+    {
+        [JsonIgnore]
+        public override string Id { get; }
+    }
+    public class CovariantDerivedGeneric<T> : CovariantBase
+        where T : class
+    {
+        [JsonIgnore]
+        public override T Id { get; }
+    }
+
+    public class IgnoredPropertyBase
```

Comment:
I'd suggest to change order of these classes to make it more obvious which cases are covered, i.e.:
- NotIgnoredPropertyBase
- NotIgnoredPropertyBase_NotIgnoredPropertyDerived
- NotIgnoredPropertyBase_IgnoredPropertyDerived
- IgnoredPropertyBase
- IgnoredPropertyBase_NotIgnoredPropertyDerived
- IgnoredPropertyBase_IgnoredPropertyDerived